pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -1315,15 +1316,18 @@ func (c *Cluster) startMon(m *monConfig, schedule *MonScheduleInfo) error {
 		}
 	}
 
-	if schedule == nil || schedule.Zone != "" {
-		k8sutil.SetNodeAntiAffinityForPod(&d.Spec.Template.Spec, requiredDuringScheduling(&c.spec), v1.LabelHostname,
-			map[string]string{k8sutil.AppAttr: AppName}, nil)
+	var nodeSelector map[string]string
+	if schedule == nil || (monVolumeClaim != nil && zone != "") {
```

Comment:
I wonder if we could name this condition to give it a more comprehensive meaning.

Reply:
Does the comment on the next line clarify? I'm not sure I could give this a variable name that isn't as long as a comment. :)